src/test/scala/projectEuler/LargestPrimeFactorCalculatorSpec.scala

```diff
@@ -2,41 +2,21 @@ package projectEuler
 
 import org.scalatest.{Matchers, WordSpec}
 
-/**
- * Created by sonia on 12/10/2015.
- */
 class LargestPrimeFactorCalculatorSpec extends WordSpec with Matchers {
 
-  "Result" when {
-    val number: Int = 13195
-    "calculate the largest prime factor of a number: " + number should {
-      val expected: Option[Long] = Some(29)
-      "return " + expected in {
-        val largestPrimeFactor = new LargestPrimeFactorCalculator(number)
-        largestPrimeFactor.result should equal(expected)
+  def testResult(description: String, number: Long, expected: Option[Long]) = {
+    "Result" when {
+       s"$description, number = $number" should {
+        "return " + expected in {
+          val largestPrimeFactor = new LargestPrimeFactorCalculator(number)
+          largestPrimeFactor.result should equal(expected)
+        }
       }
     }
   }
 
-  "Result" when {
-    val number: Int = 29
-    "calculate the largest prime factor of a prime number: " + number should {
-      val expected: Option[Long] = None
-      "return " + expected in {
-        val largestPrimeFactor = new LargestPrimeFactorCalculator(number)
-        largestPrimeFactor.result should equal(expected)
-      }
-    }
-  }
+  testResult("calculate the largest prime factor of a number", 13195L, Some(29))
+  testResult("calculate the largest prime factor of a prime number", 29L, None)
+  testResult("calculate the largest prime factor of an outbound number", 1L, None)
```

Comment:
I think this was a special case that should return None. Maybe, I should describe like "calculate the largest prime factor of 1"
However, if I want to verify that is not zero or a negative number,  I can add a test in which case it should throw the exception